### ISSUE TITLE
Return focus after opening command palette

### ISSUE THREAD

**Issue by stefanv** [jupyterlab/jupyterlab#3760]
From inside an open notebook, I can bring up the command palette with a keyboard shortcut.  But, if I then decide *not* to execute a command, there doesn't seem to be a keyboard shortcut for closing the palette and returning to the notebook (escape doesn't work, a mouse click seems to be required).

**Comment by blink1073** [comment]
Good call, we should activate the "current" widget in the main area when the side panel is closed with that command.  I wonder if we should also give focus when the side panel is closed with a mouse click...

**Comment by jasongrout** [comment]
Note that closing the command palette with `Ctrl B` (`Cmd B` on mac) *does* return focus to the notebook.

**Comment by blink1073** [comment]
Ah, good call. @stefanv, were you referring exclusively to mouse-based interaction?

**Comment by jasongrout** [comment]
> The workaround suggested (Ctrl B) closes the area, but still does not return focus to the notebook, at least not on my Linux + Firefox.

Remember that the notebook will be in command mode (so won't necessarily look like it is in focus). Just checking - after doing Ctrl B, you can't press Enter to start editing a cell, for example? (This also assumes that the notebook had focus before pressing the keyboard shortcut for the command palette).

**Comment by jasongrout** [comment]
> Ideally, ESC should be bound to return to notebook inside the command palette, that is the intuitive expectation.

Before, we've had ESC close the command palette, but that was confusing. Having ESC focus the active widget does indeed sound like a more natural action.

Another thing ESC could do is clear the current command palette command, if there is anything typed (i.e., clear the palette filter), then another ESC could refocus the current work area widget.

**Comment by jasongrout** [comment]
@stefanv - are you testing with the latest jlab release (0.31.2)? We did have some focus-related fixes recently.

**Comment by stefanv** [comment]
@jasongrout I was in editing mode when I triggered the palette, so I guess I expected to be returned to that mode.  It does, indeed, return me to command mode in the notebook when I press Ctrl-B.

**Comment by tqfjo** [comment]
> Ideally, ESC should be bound to return to notebook inside the command palette, that is the intuitive expectation.

This is also how `notebook` does it

**Comment by seeM** [comment]
@jasongrout This is still an issue on the latest versions:

```sh
$ jupyter --version
Selected Jupyter core packages...
IPython          : 8.8.0
ipykernel        : 6.19.4
ipywidgets       : 8.0.4
jupyter_client   : 7.4.8
jupyter_core     : 5.1.2
jupyter_server   : 2.5.0
jupyterlab       : 4.0.0b2
nbclient         : 0.7.2
nbconvert        : 7.2.7
nbformat         : 5.7.1
notebook         : 7.0.0b0
qtconsole        : 5.4.0
traitlets        : 5.8.0
```

IMO this is a bug. Here's how to reproduce (I am on macOS):

- Open a notebook. Confirm that you can select previous/next cell and enter/exit command mode with the keyboard.
- Open the command palette with `Accel Shift C`, then close it with `Escape`.
- Almost no keyboard shortcuts work anymore! The only way I've found to refocus the notebook editor is pressing `Accel B` twice to hide/show the left area, which is an okay workaround but far from ideal.

**Comment by seeM** [comment]
Also worth noting that since there is no way to toggle left area in Jupyter Notebook v7, the only way there is to leave the keyboard and click on the notebook.

**Comment by MMesch** [comment]
[Here](https://github.com/jupyterlab/jupyterlab/compare/main...MMesch:jupyterlab:fix-command-palette-focus-simple?expand=1) is a dirty hack that fixes this issue *in dev mode* and for the *modal* command palette only. It's not a real solution though because the app objected is improperly injected/accessed and because it is unlikely that the command palette is the right place to store a history of recently active tabs.

On the other hand, I'm not experienced enough to know how to approach this properly. Could someone more knowledgeable give a hint?

**Comment by krassowski** [comment]
> because the app objected is improperly injected/accessed

Yes, you would want to pass the shell reference down to palette constructor instead. We have dedicated tracker objects but these are per widget type, unfortunately there is no general public tracker for all widget types outside of the shell object.

> because it is unlikely that the command palette is the right place to store a history of recently active tabs

This is actually fine I think.
